### PR TITLE
[full-ci] [tests-only] Modified tests so sharee can check for received shares

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
Modified Scenarios:
- Scenario Outline: send share invitation to user with different roles
- Scenario Outline: send share invitation to group with different roles

Added step to check the existence of share by the recipient.

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/7947

## How Has This Been Tested?

- locally
- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/shareInvitations.feature [hunk 1]
```diff
@@ -1111,3 +1111,29 @@ Feature: Send a sharing invitations
       | Co Owner         | folder        | FolderToShare  |
       | Uploader         | folder        | FolderToShare  |
       | Manager          | folder        | FolderToShare  |
+
+
+  Scenario Outline: send share invitation to normal user
+    And user "Alice" has created folder "FolderToShare"
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/shareInvitations.feature:1117]:
```suggestion
    Given user "Alice" has created folder "FolderToShare"
```

File: tests/acceptance/features/apiSharingNg/shareInvitations.feature [hunk 2]
```diff
@@ -1111,3 +1111,29 @@ Feature: Send a sharing invitations
       | Co Owner         | folder        | FolderToShare  |
       | Uploader         | folder        | FolderToShare  |
       | Manager          | folder        | FolderToShare  |
+
+
+  Scenario Outline: send share invitation to normal user
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/shareInvitations.feature:1116]: we should have this scenario already covered, right?

Comment (grgprarup) [reply tests/acceptance/features/apiSharingNg/shareInvitations.feature:1116]: This scenario checks whether the `sharee` receives the share or not. A new API endpoint `shareWithMe` is used for the `then` step. Previously added tests check for request and response only.

Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/shareInvitations.feature:1116]: Then maybe we can just extend the scenarios above with one Then step

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 3]
```diff
@@ -424,4 +425,34 @@ public function userRemovesSharePermissionOfAResourceInLinkShareUsingGraphAPI(
...
+		$rows = $table->getRows();
+		$response = GraphHelper::getSharesSharedWithMe(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$user,
+			$this->featureContext->getPasswordForUser($user)
+		);
+		$contents = \json_decode($response->getBody()->getContents(), true);
+
+		$fileFound = !empty(array_intersect(array_map(fn ($row) => trim($row[0], '/'), $rows), array_column($contents['value'], 'name')));
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:450]: if we have multiple files to check, this should help
```suggestion
		$fileFound = empty(array_diff(array_map(fn ($row) => trim($row[0], '/'), $rows), array_column($contents['value'], 'name')));
```

Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:450]: Will it work if there are multiple items to check for?

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

File: tests/acceptance/features/bootstrap/SharingNgContext.php (hunk 3 above)
Comment (grgprarup) [reply tests/acceptance/features/bootstrap/SharingNgContext.php]: It will work for multiple items. I have updated array_intersect with array_diff.

**Pull request merged**